File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.cpp
#include <string>
#include <vector>
#include <map>
using namespace std;

int solution(vector<int> rank, vector<bool> attendance) {
    int answer = 0;
    int mul = 10000;
    map<int, bool> rankMap;
    map<int, int> indexMap;
    for(int i=0;i<rank.size();i++){
        rankMap.insert(make_pair(rank[i], attendance[i]));
        indexMap.insert(make_pair(rank[i], i));
    }
    map<int, bool>::iterator iter;
    
    for (iter = rankMap.begin(); iter != rankMap.end(); iter++) {
        if(iter->second){
            answer = answer + (mul * indexMap[iter->first]);
            mul /= 100;
        }
        if(mul==0) break;
        
    } 
    return answer;
}

File: 프로그래머스/0/181851. 전국 대회 선발 고사/전국 대회 선발 고사.java
import java.util.*;

class Solution {
    public int solution(int[] rank, boolean[] attendance) {
        int answer = 0;
        int score = 10000;
        SortedMap<Integer, Boolean> rankMap = new TreeMap<Integer, Boolean>();
        SortedMap<Integer, Integer> indexMap = new TreeMap<Integer, Integer>();
        
        for(int i=0;i<rank.length;i++){
            rankMap.put(rank[i], attendance[i]);
            indexMap.put(rank[i], i);
        }
        
        for(int i=1;i<=rank.length;i++){
            if(rankMap.get(i)){
                answer = answer + (score * indexMap.get(i));
                score /=100;
            }
            if(score == 0) break;
        }
        
        
        return answer;
    }
}